santa/santa.munki.recipe

```diff
@@ -78,43 +158,41 @@
 				<dict>
 					<key>version</key>
 					<string>%version%</string>
-				        <key>installcheck_script</key>
-				        <string>#!/usr/local/munki/munki-python
-import plistlib
-import sys
-
-
-def test_is_santa(extension):
-    return (
-        extension.get("identifier") == "com.northpolesec.santa.daemon"
-        and extension.get("teamID") == "ZMCG7MLDV9"
-    )
+					<key>installcheck_script</key>
+					<string>#!/bin/zsh
 
+minimum="%version%"
+SANTACTL="/Applications/Santa.app/Contents/MacOS/santactl"
+INFOPLIST="/Applications/Santa.app/Contents/Info.plist"
+PLISTBUDDY="/usr/libexec/PlistBuddy"
 
-def get_version(extension):
-    return extension.get("bundleVersion", {}).get("CFBundleShortVersionString")
+# Make sure santactl exists and is executable
+if [[ ! -x "${SANTACTL}" ]]; then
+	echo "santactl not found or not executable"
+	exit 0
+fi
 
+# Get Santa version
+current="$("${PLISTBUDDY}" -c "Print :CFBundleVersion" "${INFOPLIST}" 2&gt;/dev/null || true)"
+if [[ -z "${current}" ]]; then
+	echo "ERROR: failed to read CFBundleVersion from ${INFOPLIST}"
+	exit 0
+fi
 
-def do_install_check():
-    with open("/Library/SystemExtensions/db.plist", "rb") as f:
-        info = plistlib.load(f)
-    for extension in info.get("extensions", []):
-        if (
-            test_is_santa(extension)
-            and get_version(extension) == "%version%"
-        ):
-            return extension.get("state") == "activated_enabled"
-    return False
+# Version comparison
+autoload -Uz is-at-least
+if ! is-at-least "${minimum}" "${current}"; then
+	echo "Santa version ${current} does not meet the minimum version requirement of ${minimum}"
+	exit 0
+fi
 
+# Make sure Santa is running
+if ! "${SANTACTL}" status &gt;/dev/null 2&gt;&amp;1; then
```

Comment:
OK, I understand that if Santactl is failing that means that we have to reinstall. But could we also fail if the versions reported by status are not the expected ones? Count with grep -c and make sure we have 3 matches for example?